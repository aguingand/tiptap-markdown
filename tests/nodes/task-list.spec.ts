import { expect, test } from "vitest";
import { parse, serialize } from "../utils";

test('parse markdown', () => {
    expect(parse('- [ ] example1\n- [x] example2')).toMatchSnapshot();
});

test('serialize', () => {
    expect(serialize('<ul data-type="taskList">' +
        '<li data-type="taskItem" data-checked="false">example1</li>' +
        '<li data-type="taskItem" data-checked="true">example2</li>' +
    '</ul>'))
        .toEqual('- [ ] example1\n- [x] example2\n')
});
