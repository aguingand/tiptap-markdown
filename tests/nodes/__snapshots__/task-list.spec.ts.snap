// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`parse markdown 1`] = `
[
  {
    "attrs": {
      "tight": true,
    },
    "content": [
      {
        "attrs": {
          "checked": false,
        },
        "content": [
          {
            "content": [
              {
                "text": "example1",
                "type": "text",
              },
            ],
            "type": "paragraph",
          },
        ],
        "type": "taskItem",
      },
      {
        "attrs": {
          "checked": true,
        },
        "content": [
          {
            "content": [
              {
                "text": "example2",
                "type": "text",
              },
            ],
            "type": "paragraph",
          },
        ],
        "type": "taskItem",
      },
    ],
    "type": "taskList",
  },
]
`;
